tests/test_pytypes.py

```diff
@@ -220,6 +220,27 @@ def test_constructors():
         assert noconv2[k] is expected[k]
 
 
+def test_pybind11_str_raw_str():
+    # specifically to exercise pybind11::str::raw_str
+    cvt = m.convert_to_pybind11_str
+    assert cvt(u"Str") == u"Str"
+    assert cvt(b"Bytes") == u"Bytes" if str is bytes else "b'Bytes'"
```

Comment:
Should we maybe use `sys.version_info.major < 3` instead of `str is bytes`?

Or start using some kind of compatibility library (cfr. https://python-future.org/compatible_idioms.html#strings-and-bytes, or `six`)? Another thing that could make things simpler/more Python 3: `from __future__ import unicode_literals` would allow us to remove all `u` prefixes.

Reply:
There were other parts of test code that explicitly check for 2 vs. 3, I believe. Let me find it.

Reply:
Examples:
https://github.com/pybind/pybind11/blob/0af7fe6c1943e6a9043e4e01c4bc9059108a6c98/tests/test_builtin_casters.py#L119
https://github.com/pybind/pybind11/blob/0af7fe6c1943e6a9043e4e01c4bc9059108a6c98/tests/test_buffers.py#L11

Reply:
FWIW I've filed #2376